openapi.yml

```diff
@@ -1253,12 +1791,421 @@ components:
+
+    Transaction:
+      type: object
+      properties:
+        siacoinInputs:
+          type: array
+          description: List of Siacoin inputs used in the transaction.
+          items:
+            $ref: "#/components/schemas/SiacoinInput"
+        siacoinOutputs:
+          type: array
+          description: List of Siacoin outputs created by the transaction.
+          items:
+            $ref: "#/components/schemas/SiacoinOutput"
+        fileContracts:
+          type: array
+          description: List of file contracts created by the transaction.
+          items:
+            $ref: "#/components/schemas/FileContract"
+        fileContractRevisions:
+          type: array
+          description: List of revisions to existing file contracts included in the transaction.
+          items:
+            $ref: "#/components/schemas/FileContractRevision"
+        storageProofs:
+          type: array
+          description: List of storage proofs asserting the storage of data for file contracts.
+          items:
+            $ref: "#/components/schemas/StorageProof"
+        siafundInputs:
+          type: array
+          description: List of Siafund inputs spent in the transaction.
+          items:
+            $ref: "#/components/schemas/SiafundInput"
+        siafundOutputs:
+          type: array
+          description: List of Siafund outputs created by the transaction.
+          items:
+            $ref: "#/components/schemas/SiafundOutput"
+        minerFees:
+          type: array
+          description: List of miner fees included in the transaction.
+          items:
+            $ref: "#/components/schemas/Currency"
+        arbitraryData:
+          type: array
+          description: Arbitrary binary data included in the transaction.
+          items:
+            type: string
+            format: byte
+        signatures:
+          type: array
+          description: List of cryptographic signatures verifying the transaction.
+          items:
+            $ref: "#/components/schemas/TransactionSignature"
+
+    TransactionID:
+      allOf:
+        - $ref: "#/components/schemas/Hash256"
+        - description: Unique identifier for a transaction.
+
+    TransactionSignature:
+      type: object
+      properties:
+        parentID:
+          allOf:
+            - $ref: "#/components/schemas/Hash256"
+            - description: The ID of the transaction being signed
+        publicKeyIndex:
+          type: integer
+          format: uint64
+          description: The index of the public key used to sign the transaction
+        timelock:
+          type: integer
+          format: uint64
+          description: The block height at which the output can be spent
+        coveredFields:
+          allOf:
+            - $ref: "#/components/schemas/CoveredFields"
+            - description: Indicates which fields of the transaction are covered by the signature
+        signature:
```

Comment:
Don't you have a signature type you can reuse that has a pattern? Right now this just says "string" in the editor.